mbed_host_tests/__init__.py

```diff
@@ -494,14 +479,25 @@ def execute(self):
                 self.notify("HOST: Error! No host test name defined in preamble")
                 self.print_result(self.RESULT_ERROR)
 
-            if result is not None:
-                self.print_result(result)
+            if self.result is not None:
+                self.print_result(self.result)
             else:
                 self.notify("HOST: Passive mode...")
         except Exception, e:
             print str(e)
             self.print_result(self.RESULT_ERROR)
 
+    def run_test(self, host_test_name):
+        try:
+            self.test_supervisor = get_host_test(host_test_name)
+            self.test_supervisor.setup()
+            self.result = self.test_supervisor.test(self)    # This is blocking, waits for {end}
```

Comment:
This function can return the result in the end. We should put less variables in the object state as they would need maintenance.